src/browser/event.replayer.ts

```diff
@@ -0,0 +1,284 @@
+import { Injectable } from '@angular/core';
+import {
+    ServerClientRoot,
+    PrebootCompleteOptions,
+    PrebootAppData,
+    PrebootData,
+    PrebootEvent,
+    Element,
+    NodeContext,
+    getNodeKeyForPreboot
+} from '../common';
+import { WindowRef } from './window';
+
+@Injectable()
+export class EventReplayer {
```

Comment:
This is very similar to the old preboot client code, but just wrapped in an Angular service.